src/cas_cli/client.py

```diff
@@ -0,0 +1,238 @@
+        """
+        number_of_cells = chunk_end_i - chunk_start_i
+        for _ in range(3):
+            try:
+                results[chunk_index] = await self.cas_api_service.async_annotate_anndata_chunk(
+                    adata_file_bytes=adata_bytes, number_of_cells=number_of_cells
+                )
+
+            except exceptions.HTTPError500:
+                self._print(
+                    f"Something went wrong, resubmitting chunk #{chunk_index + 1:2.0f} ({chunk_start_i:5.0f}, {chunk_end_i:5.0f}) to CAS ..."
+                )
+                pass
+            except exceptions.HTTPError401:
+                self._print("Unauthorized token. Please check your API token or request a new one.")
+                break
+            else:
+                self._print(
+                    f"Received the annotations for cell chunk #{chunk_index + 1:2.0f} ({chunk_start_i:5.0f}, {chunk_end_i:5.0f}) ..."
+                )
+                break
+
+    async def _annotate_anndata_task(self, adata, results, chunk_size, start_time) -> None:
+        """
+        Submit chunks asynchronously as asyncio tasks
+
+        :param adata: AnnData file to process
+        :param results: Results list that is used to inplace the responses from the server
+        :param chunk_size: Chunk size to split on
+        :param start_time: Start time of the execution before this task was submitted
+        """
+        i, j = 0, chunk_size
+        tasks = []
+        number_of_chunks = self._get_number_of_chunks(adata, chunk_size=chunk_size)
+        for chunk_index in range(number_of_chunks):
+            chunk = adata[i:j, :]
+            chunk_start_i = i
+            chunk_end_i = i + len(chunk)
+            self._print(
+                f"Submitting cell chunk #{chunk_index + 1:2.0f} ({chunk_start_i:5.0f}, {chunk_end_i:5.0f}) to CAS ..."
+            )
+            tmp_file_name = f"chunk_{chunk_index}.h5ad"
+            chunk.write(tmp_file_name, compression="gzip")
+            with open(tmp_file_name, "rb") as f:
+                chunk_bytes = f.read()
+
+            os.remove(tmp_file_name)
+            tasks.append(
+                self._annotate_anndata_chunk(
+                    adata_bytes=chunk_bytes,
+                    results=results,
+                    chunk_index=chunk_index,
+                    chunk_start_i=chunk_start_i,
+                    chunk_end_i=chunk_end_i,
+                )
+            )
+            i = j
+            j += chunk_size
+
+        await asyncio.wait(tasks)
+
+    def annotate_anndata(
+        self,
+        adata: "anndata.AnnData",
+        chunk_size=2000,
+        feature_schema_name: str = "default",
+        count_matrix_name: str = "X",
+        feature_ids_column_name: str = "index",
+    ) -> t.List[t.Dict[str, t.Any]]:
+        """
+        Send an anndata object to Cellarium Cloud backend to get annotations. Split the input
```

Comment:
here and everywhere in the docstrings, please replace `anndata` with `anndata.AnnData`.